Add with-edgedb example (#35929)
vercel/next.js: examples/with-edgedb/README.md

## Deploy your own

Deploy the example using [Vercel](https://vercel.com?utm_source=github&utm_medium=readme&utm_campaign=next-example):

[![Deploy with Vercel](https://vercel.com/button)](https://vercel.com/new/git/external?repository-url=https://github.com/vercel/next.js/tree/canary/examples/with-edgedb&project-name=with-edgedb&repository-name=with-edgedb)

## How to use

### Download the example project

Execute [`create-next-app`](https://github.com/vercel/next.js/tree/canary/packages/create-next-app) with [npm](https://docs.npmjs.com/cli/init) or [Yarn](https://yarnpkg.com/lang/en/docs/cli/create/) to bootstrap the example:

```bash
npx create-next-app --example with-edgedb with-edgedb-app
# or
yarn create next-app --example with-edgedb with-edgedb-app
# or
pnpm create next-app -- --example with-edgedb with-edgedb-app
```

Then `cd` into the created directory.

```bash
$ cd with-edgedb-app
```

### Install dependencies

Install npm dependencies:

```bash
$  npm install
# or
$  yarn
```

### Install the CLI

First install the EdgeDB CLI if you haven't already.

```bash
# macOS/Linux
$ curl --proto '=https' --tlsv1.2 -sSf https://sh.edgedb.com | sh

> # windows

Reviewer: ```suggestion
# windows (powershell)
```

Author: 👍

- `PUT /api/publish/:id`: Publish a post by `id`
- `DELETE /api/post/:id`: Delete a post by `id`

## Evolving the app

Evolving the application typically requires three steps:

1. Update the schema in `dbschema/default.esdl`
1. Generate a new migration with `edgedb migration create`
1. Apply the migration with `edgedb migrate`
1. Regenerate the query builder with `npx edgeql-js`

## Deployment

To deploy this application, follow the EdgeDB deployment guides for your preferred cloud provider:

- [AWS](https://www.edgedb.com/docs/guides/deployment/aws_aurora_ecs)
- [Google Cloud](https://www.edgedb.com/docs/guides/deployment/gcp)
- [Azure](https://www.edgedb.com/docs/guides/deployment/azure_flexibleserver)
- [DigitalOcean](https://www.edgedb.com/docs/guides/deployment/digitalocean)
- [Fly.io](https://www.edgedb.com/docs/guides/deployment/fly_io)
- [Docker](https://www.edgedb.com/docs/guides/deployment/docker) (cloud-agnostic)

After setting up your cloud-hosted EdgeDB instance:

1. Deploy this app to Vercel with the button below.
2. Find your instance's DSN (AKA connection string). The exact instructions for this depend on which cloud you are deploying to.

> 3. Set the value of `EDGEDB_DSN` environment variable via the Vercel dashboard.

Reviewer: Seems like this variable should be set before the first deploy, it can be set during the project creation flow by changing the URL like below

Author: Updated

> 2. Deploy this app to Vercel with the button below. You'll be prompted to provide a value for `EDGEDB_DSN`.

Reviewer: ```suggestion
2. Deploy this app to Vercel with the [button above](#deploy-your-own). You'll be prompted to provide a value for `EDGEDB_DSN`, this is the value from the previous step.
```

> [![Deploy with Vercel](https://vercel.com/button)](https://vercel.com/new/git/external?repository-url=https://github.com/vercel/next.js/tree/canary/examples/with-edgedb&project-name=with-edgedb&repository-name=with-edgedb)

Reviewer: ```suggestion
[![Deploy with Vercel](https://vercel.com/button)](https://vercel.com/new/git/external?repository-url=https://github.com/vercel/next.js/tree/canary/examples/with-edgedb&project-name=with-edgedb&repository-name=with-edgedb&env=EDGEDB_DSN)
```

Author: Ooh, this is nice

> [![Deploy with Vercel](https://vercel.com/button)](https://vercel.com/new/git/external?repository-url=https://github.com/vercel/next.js/tree/canary/examples/with-edgedb&project-name=with-edgedb&repository-name=with-edgedb&env=EDGEDB_DSN)

Reviewer: ```suggestion
```
I don't think we want to have the deploy button twice in the README as we only have it once in our other examples.
